### DISCUSSION TITLE
How do I refresh the client so the UI updates

### DISCUSSION THREAD

**Discussion by alex-r89** [supabase/supabase#13895] [Questions] [answered]
Hi,

I have plugged in supabase to authenticate my users and all seems to be working fine. The issue I am facing though is that UI is not updating to reflect that a user has just logged in/out. A user has to manually refresh the page to update the UI.

I am using `signInWithOtp` and the url seems to work fine, it just requires a manual refresh to then update the UI.

I have the following (simplified to remove irrelevant parts) code:

### _app.jsx
```js
import React, { useState } from 'react'
import { createBrowserSupabaseClient } from '@supabase/auth-helpers-nextjs'
import { SessionContextProvider } from '@supabase/auth-helpers-react'

export default function App(props) {
  const { Component, pageProps } = props
  const [supabaseClient] = useState(() => createBrowserSupabaseClient())

  return (
      <SessionContextProvider
        supabaseClient={supabaseClient}
        initialSession={pageProps.initialSession}>
            <Component {...pageProps} />
      </SessionContextProvider>
  )
}

```
Then, within pages, I access the user like so: 
```js
export const getServerSideProps = async (ctx) => {
  const supabase = createServerSupabaseClient(ctx)
  const {
    data: { session }
  } = await supabase.auth.getSession()

  if (!session)
    return {
      props: {}
    }

  return {
    props: {
      initialSession: session,
      user: session.user
    }
  }
}
```

I understand this is a server side request for the user, so I assume there is something more I need to add to the app.jsx file in order to refresh the client?

I have tried using  `useUser` from '@supabase/auth-helpers-react', in place of the server side `getServerSideProps` request, but doing this seems to result in a UI flicker.

I have also tried adding in a `supabaseClient.auth.refreshSession()` in the _app.jsx file but this didnt seem to work/I guess I did it wrong.

Can anyone advise on how to update the UI on the client once a user successfully logs in?

Thanks

**Comment by j4w8n** [comment]
Did you get this working? I don't know this framework; but I do know that if you use `refreshSession()` on the server-side you need to pass in an object with the access_token and refresh_token. I don't know if the react auth helper is supposed to help with that.

**Comment by Ziinc** [comment, marked as answer]
You can use the `supabase.auth.onAuthStateChange()` function to subscribe to auth changes on the client. You cna then use a useEffect to handle the auth changes

https://supabase.com/docs/reference/javascript/auth-onauthstatechange
